libdleyna/core/task-processor.c

```diff
@@ -202,9 +202,8 @@ static gboolean prv_cancel_only(const dleyna_task_queue_key_t *queue_id,
 	if (task_queue->current_task)
 		task_queue->task_cancel_cb(task_queue->current_task,
 					   task_queue->user_data);
-	else
-		remove_queue = task_queue->flags &
-			DLEYNA_TASK_QUEUE_FLAG_AUTO_REMOVE;
+
+	remove_queue = task_queue->flags & DLEYNA_TASK_QUEUE_FLAG_AUTO_REMOVE;
```

Comment:
This isn't actually preventing the `g_hash_table_remove`, right? Because the callback can still return `TRUE`.

I think the original code here was OK, because if we remove the queue for the running task then ...